### PR TITLE
feat(browser): improve worker_threads 

### PR DESCRIPTION
**Pull request by Mesteery**
*This has not been tested but works correctly on Deno, and therefore surely also on browsers.*

### CODE REVIEW

File: src-browser/worker_threads.js [hunk 1]
```diff
@@ -18,31 +19,40 @@ export class Worker extends globalThis.Worker {
...
 
   constructor(specifier, options) {
-    super(specifier, { ...(options || {}), type: 'module' });
-    EventEmitter.call(this);
-    this.addEventListener('error', (event) => this.emit('error', event.error || event.message));
-    this.addEventListener('messageerror', (event) => this.emit('messageerror', event.data));
-    this.addEventListener('message', (event) => this.emit('message', event.data));
-    this.postMessage({
+    super();
+    if (options?.eval === true) {
```
Comment (guybedford) [inline src-browser/worker_threads.js:23]: Avoid optional chaining:

```suggestion
    if (typeof options === 'object' && options !== null && options.eval === true) {
```

File: src-browser/worker_threads.js [hunk 2]
```diff
@@ -18,31 +19,40 @@ export class Worker extends globalThis.Worker {
...
+      specifier = `data:text/javascript,${specifier}`;
+    }
+    const handle = this[kHandle] = new globalThis.Worker(specifier, {
+      ...(options || {}),
+      type: 'module',
+      // unstable
+      deno: { namespace: true },
+    });
```
Comment (guybedford) [inline src-browser/worker_threads.js:31]: Avoid `globalThis` and object spread:

```suggestion
    const _global = typeof self !== 'undefined' ? self : typeof window !== 'undefined' ? window : typeof global !== 'undefined' ? global : null;
    const handle = this[kHandle] = Object.assign(new _global.Worker(specifier, Object.assign({}, options || {}, {
      type: 'module',
      // unstable
      deno: { namespace: true },
    }));
```

**Review by guybedford — COMMENTED** [review]
See the syntax comments.

File: src-browser/worker_threads.js [hunk 3]
```diff
@@ -18,31 +19,40 @@ export class Worker extends globalThis.Worker {
...
 
   terminate() {
-    super.terminate();
+    this[kHandle].terminate();
     this.emit('exit', 0);
   }
 
   getHeapSnapshot = () => unimplemented('Worker#getHeapsnapshot');
   // fake performance
   performance = globalThis.performance;
```
Comment (guybedford) [inline src-browser/worker_threads.js:51]:
```suggestion
  performance = _global.performance;
```

File: src-browser/worker_threads.js (hunk 2 above)
Comment (Mesteery) [reply src-browser/worker_threads.js]: Is there a reason for the `_global`?

Comment (guybedford) [reply src-browser/worker_threads.js]: `globalThis` is a recent browser addition.

File: src-browser/worker_threads.js [hunk 4]
```diff
@@ -18,31 +21,39 @@ export class Worker extends globalThis.Worker {
...
```
Comment (guybedford) [inline src-browser/worker_threads.js:52]:
```suggestion
  performance = _global.performance;
```

**Pull request merged**